cloudimg/aws.py

```diff
@@ -40,18 +57,22 @@ class AWSPublishingMetadata(PublishingMetadata):
                                            interfaces. By default this option
                                            is enabled.
         billing_products (list, optional): Billing product identifiers
+
+        boot_mode (str, optional): The boot mode for booting up the AMI on EC2.
     """
 
     def __init__(self, *args, **kwargs):
         self.ena_support = kwargs.pop('ena_support', True)
         self.sriov_net_support = kwargs.pop('sriov_net_support', 'simple')
         self.billing_products = kwargs.pop('billing_products', None)
+        self.boot_mode = AWSBootMode(kwargs.pop('boot_mode', None))
 
         super(AWSPublishingMetadata, self).__init__(*args, **kwargs)
 
-        # Convert uefi_support bool into AWS specific values
-        self.boot_mode = "uefi-preferred" if self.uefi_support \
-            else "legacy-bios"
+        # Convert uefi_support bool into AWS specific value when
+        #  boot_mode is not set and UEFI support is True.
+        if self.boot_mode == AWSBootMode.not_set and self.uefi_support is True:
+            self.boot_mode = AWSBootMode.hybrid
```

Comment:
we would not set `"legacy-bios"` as it was before.
```
if self.boot_mode == AWSBootMode.not_set and self.uefi_support is not None:
     self.boot_mode = AWSBootMode.hybrid if self.uefi_support else AWSBootMode.bios
```
or we might enhance code and log warning when both `boot_mode` and `uefi_support` are passed as input. e,g, `uefi_support` is ignored because `boot_mode` is detected.

Reply:
Cool, I liked this condition, I'll adjust for it

Reply:
tyvm